python/lsst/ts/atpneumaticssimulator/pneumatics_simulator.py

```diff
@@ -239,6 +387,179 @@ async def write_noack_response(self, sequence_id: int) -> None:
     async def write_success_response(self, sequence_id: int) -> None:
         await self.write_response(Ack.SUCCESS, sequence_id)
 
+    async def _write_evt(self, evt_id: str, **kwargs: typing.Any) -> None:
+        await self.cmd_evt_server.write_json(data={"id": evt_id, **kwargs})
+
+    async def _write_telemetry(self, tel_id: str, data: typing.Any) -> None:
+        # This needs to be cleaned up as soon as we have moved to Kafka.
+        data_dict = data.get_vars() if hasattr(data, "get_vars") else vars(data)
+        items = {k: v for k, v in data_dict.items() if not k.startswith("private")}
+
+        items["id"] = tel_id
+        await self.telemetry_server.write_json(data={**items})
+
+    async def cmd_evt_connect_callback(self, server: tcpip.OneClientServer) -> None:
+        """Callback function for when a cmd/evt client connects or disconnects.
+
+        When a cmd/evt client connects, background tasks are started and events
+        are sent.
+        When the cmd/evt client disconnects, all background tasks get stopped.
+        """
+        if server.connected:
+            await self.configure()
+            await self.initialize()
+
+    async def tel_connect_callback(self, server: tcpip.OneClientServer) -> None:
+        """Callback function for when a tel client connects or disconnects.
+
+        When a tel client connects, the telemetry loop is started.
+        When the tel client disconnects, the telemetry loop is stopped.
+        """
+        if server.connected:
+            if self._telemetry_task.done():
+                self._telemetry_task = asyncio.create_task(self.telemetry_loop())
+        else:
+            self._telemetry_task.cancel()
+
+    async def set_cell_vents_events(self, closed: bool, opened: bool) -> None:
+        """Set m1VentsLimitSwitches, m1VentsPosition and cellVentsState events.
+
+        Output any changes.
+
+        Parameters
+        ----------
+        closed : `bool`
+            Are the closed switches active?
+        opened : `bool`
+            Are the opened switches active?
+        """
+        assert not (closed and opened)
```

Comment:
I suggest a more informative error message if this happens. Perhaps RuntimeError (though you can include a string on an assert expression if you are keen to use that)

Reply:
I wasn't happy with this assert (which I copied one on one from the original CSC simulator code) since it basically is ineffective. The simulator needs to send a FAIL message to the CSC in case both closed and opened are True and that would involve a lot of housekeeping since the method gets called in various places. However, I checked the code and in none of the cases where this method is called, it is called with both set to True so I removed the assert.